Mathlib/Data/Set/Lattice.lean

```diff
@@ -2300,3 +2300,19 @@ lemma forall_sUnion {S : Set (Set α)} {p : α → Prop} :
 lemma exists_sUnion {S : Set (Set α)} {p : α → Prop} :
     (∃ x ∈ ⋃₀ S, p x) ↔ ∃ s ∈ S, ∃ x ∈ s, p x := by
   simp_rw [← exists_prop, ← iSup_Prop_eq, iSup_sUnion]
+
+theorem sInterIio_eq_sInterIio_inter_Ico {α β : Type _} [LinearOrder β] {X : β → Set α} {D N : β}
+    (hND : N ≤ D) : (⋂ d ∈ Iio D, X d) = (⋂ d ∈ Iio N, X d) ∩ ⋂ d ∈ Ico N D, X d := by
+  by_cases hND₀ : N = D
+  · simp only [mem_Iio, hND₀, gt_iff_lt, lt_self_iff_false, not_false_eq_true, Ico_eq_empty,
+      mem_empty_iff_false, iInter_of_empty, iInter_univ, inter_univ]
+  · replace hND := lt_of_le_of_ne hND hND₀
+    rw [← iInter_inter_distrib, ← max_eq_right (le_refl D), ← Iio_union_Ioo (min_lt_of_left_lt hND),
+      max_eq_right (le_refl D)]
+    congr with d
+    simp only [mem_union, mem_Iio, mem_Ico, mem_Ioo, mem_iInter, mem_inter_iff, and_imp]
+    refine ⟨fun h b => ⟨fun H => h b <| Or.inl <| H.trans hND, fun _ h_ND => h b <| Or.inl h_ND⟩,
+      fun h b H => ?_⟩
+    rcases H with (Ha | Hb)
+    by_cases H_bN : b < N
+    exacts [(h b).1 H_bN, (h b).2 (le_of_not_lt H_bN) Ha, (h b).2 (le_of_lt Hb.1) Hb.2]
```

Comment:
```suggestion
  rw [← biInter_union, Iio_union_Ico_eq_Iio hND]
```

I would say that this is short enough that we don't need the lemma, but I'm ready to be convinced otherwise.

Reply:
Wow, that was nice! I completely overlooked this, thanks (and I realise that two shots of `rw?` do the game: did you find it this way or did you imagine what to do?). I am going to close the PR, at any rate, I agree we do not need it.

Reply:
Oh no, using `rw?` ruins all the fun and the satisfaction of knowing my way around (parts of) the library!

Reply:
Chapeau!